warcprox/dedup.py

```diff
@@ -200,37 +201,36 @@ def save(self, digest_key, response_record, bucket=""):
         """
         pass
 
-    def lookup(self, digest_key, recorded_url):
+    def lookup(self, digest_key, url):
         """Compare `sha1` with SHA1 hash of fetched content (note SHA1 must be
         computed on the original content, after decoding Content-Encoding and
         Transfer-Encoding, if any), if they match, write a revisit record.
 
         :param digest_key: b'sha1:<KEY-VALUE>' (prefix is optional).
             Example: b'sha1:B2LTWWPUOYAH7UIPQ7ZUPQ4VMBSVC36A'
-        :param recorded_url: RecordedUrl object
+        :param url: Target URL string
         Result must contain:
         {"url": <URL>, "date": "%Y-%m-%dT%H:%M:%SZ"}
         """
-        url = recorded_url.url
         u = url.decode("utf-8") if isinstance(url, bytes) else url
         try:
             result = self.http_pool.request('GET', self.cdx_url, fields=dict(
                 url=u, fl="timestamp,digest", filter="!mimetype:warc/revisit",
-                limit=-10))
+                limit=-1))
             assert result.status == 200
             if isinstance(digest_key, bytes):
                 dkey = digest_key
             else:
                 dkey = digest_key.encode('utf-8')
             dkey = dkey[5:] if dkey.startswith(b'sha1:') else dkey
-            for line in result.data.split(b'\n'):
-                if line:
-                    (cdx_ts, cdx_digest) = line.split(b' ')
-                    if cdx_digest == dkey:
-                        dt = datetime.strptime(cdx_ts.decode('ascii'),
-                                               '%Y%m%d%H%M%S')
-                        date = dt.strftime('%Y-%m-%dT%H:%M:%SZ').encode('utf-8')
-                        return dict(url=url, date=date)
+            line = result.data.split(b'\n')
```

Comment:
We're definitely down to the nitpicky stuff now, but this is weird. Splitting on newline when there is never more than one line in the results. Can I suggest `line = result.data.strip()```

Reply:
Yes, good point, I agree.